addons/mrp/tests/test_consume_tracked_component.py

```diff
@@ -0,0 +1,364 @@
+from odoo.exceptions import UserError
+from odoo.tests import common, Form, tagged
+import copy
+
+SERIAL_AVAILABLE_TRIGGERS_COUNT = 3
+DEFAULT_AVAILABLE_TRIGGERS_COUNT = 2
+SERIAL_TRIGGERS_COUNT = 2
+DEFAULT_TRIGGERS_COUNT = 1
+
+@tagged('post_install', '-at_install')
+class TestConsumeTrackedComponent(common.TransactionCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.manufacture_route = cls.env.ref('mrp.route_warehouse0_manufacture')
+        cls.stock_id = cls.env.ref('stock.stock_location_stock').id
+
+        cls.picking_type = cls.env['stock.picking.type'].search([('code', '=', 'mrp_operation')])[0]
```

Comment:
You got it right for `self.registry.reset_changes()` so I just removed them.
Problem with the `generate_mo` method is that it only allows 2 components, but in my case I want to test the behavior on each tracking type, such that I need 3 components.
As you say, refactoring of the TestMrpCommon may be part of another PR, but for sure, these variable could be defined there. In the same idea, we can imagine refactoring the `generate_mo `method to receive dictionary in parameters so that it can generate MO with an arbitrary number of components and an user-defined component type (such as consumable as it may be used in some tests).